docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc

```diff
@@ -0,0 +1,64 @@
+= Composition Function Error Handling
+
+In March 2024 we had a severe outage of VSHN PostgreSQL instances.
+This was due to several unfortunate events including our poor handling of errors in composition functions.
+This document will discuss in detail the problem we face with composition functions when errors are not properly handled.
+
+== Problems
+
+* We don't manage errors in our composition functions (CFs) and we assume Crossplane will take care of everything.
+* Missing managed resources from desired function when an unexpected error happens can have dire consequences, such as removal of the whole service.
+* Returning early from composition function (CF) due to a warning can lead to missing managed resources.
+
+== Requirements
+
+* A comprehensive guide on how we should deal with error management.
+* A solution to missing managed resources from desired function when errors occur.
+
+== Proposals
+
+Proposal 1:: Manage Errors in CFs
+
+We don't throw fatal errors but use mostly warnings in our composition functions.
+Using warnings and return right away can remove vital parts of a service.
+In order to mitigate this issue we could define the return error (warning or fatal) directly in the runtime.
```

Comment:
For the sake of the decision: quickly explain the difference between warning and fatal and the implications. Maybe not everyone knows the difference.

Reply:
Added Definition section